### PR TITLE
Add hint for oneshot property & warning when it will be updated continuously by Force Continuous in `AnimationMixer`

### LINKED ISSUE #95684: AnimationTree state machine causes glitchy sound when triggering `playing` of AudioStreamPlayer3D
### Tested versions

Reproducible in 4.3, _not reproducible_ in 4.2

### System information

Godot v4.3.stable unknown - Arch Linux #1 SMP PREEMPT_DYNAMIC Sun, 11 Aug 2024 16:19:06 +0000 - Wayland - GLES3 (Compatibility) - AMD Radeon RX 6800 XT (radeonsi, navi21, LLVM 18.1.8, DRM 3.57, 6.10.4-arch2-1) - AMD Ryzen 7 5800X3D 8-Core Processor (16 Threads)

### Issue description

If you have an AnimationTree with a state machine that triggers the `playing` property on an AudioStreamPlayer3D in a keyframe, the audio becomes glitched, sounding as if it is constantly restarting from the first sample every frame. Just load up the MRP to see it in action; works fine on Godot 4.2, glitches audio in Godot 4.3.

### Steps to reproduce

play the MRP

### Minimal reproduction project (MRP)

[animtree_audio.zip](https://github.com/user-attachments/files/16643723/animtree_audio.zip)

### PR DESCRIPTION
**Pull request by TokageItLab**
- Closes https://github.com/godotengine/godot/issues/95684

These properties should not be continually updated, so that warnings are issued when they may occur unintentionally.

Ideally, FLAG_USAGE should be used, but I fear the int value limit as flag, so I append that to HINT;

As I remember only audio and partcle bool value properties should have these, but if there are non-bool value properties in other classes that should have this, they should be in USAGE. If there is such a thing, please let me know.

### CODE REVIEW

**Review by fire — APPROVED** [review]

**Comment by KoBeWi** [comment]
This should really be a property usage, but I think we can add just only one more usage value🤔
Then again, none of the current ONE_SHOT properties have problem with this being a hint, so we could treat it as temporary solution and think of a fix once an actual problem arises (i.e. conflict with another hint, as properties can't have more than 1 hint). To avoid breaking compatibility in the future, you can unexpose the new hint, with a comment that it shouldn't be exposed, because it's improper solution or something.

File: scene/animation/animation_mixer.cpp [hunk 1]
```diff
@@ -716,6 +716,15 @@ bool AnimationMixer::_update_caches() {
 							}
 						}
 
+						if (is_value && callback_mode_discrete == ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS) {
+							if (child) {
+								PropertyInfo prop_info;
+								ClassDB::get_property_info(child->get_class_name(), path.get_concatenated_subnames(), &prop_info);
+								if (prop_info.hint == PROPERTY_HINT_ONESHOT) {
+									WARN_PRINT_ED(mixer_name + ": '" + String(E) + "', Value Track: '" + String(path) + "' is oneshot property, but will be continuously updated. Consider setting a value other than ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS to AnimationMixer.callback_mode_dominant.");
```
Comment (AThousandShips) [inline scene/animation/animation_mixer.cpp:724]:
```suggestion
									WARN_PRINT_ED(vformat("%s: '%s', Value Track: '%s' is oneshot property, but will be continuously updated. Consider setting a value other than ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS to AnimationMixer.callback_mode_dominant.", mixer_name, String(E), String(path)));
```

File: doc/classes/@GlobalScope.xml [hunk 2]
```diff
@@ -2941,7 +2941,10 @@
 			[/codeblock]
 			[b]Note:[/b] A [Callable] cannot be properly serialized and stored in a file, so it is recommended to use [constant PROPERTY_USAGE_EDITOR] instead of [constant PROPERTY_USAGE_DEFAULT].
 		</constant>
-		<constant name="PROPERTY_HINT_MAX" value="40" enum="PropertyHint">
+		<constant name="PROPERTY_HINT_ONESHOT" value="40" enum="PropertyHint">
+			Hints that a property will be changed by self after setting, such as [member AudioStreamPlayer.playing] or [member GPUParticles3D.emitting].
```
Comment (AThousandShips) [inline doc/classes/@GlobalScope.xml:2945]:
```suggestion
			Hints that a property will be changed on its own after setting, such as [member AudioStreamPlayer.playing] or [member GPUParticles3D.emitting].
```
or
```suggestion
			Hints that a property will be changed by itself after setting, such as [member AudioStreamPlayer.playing] or [member GPUParticles3D.emitting].
```

**Review by AThousandShips — APPROVED** [review]

**Comment by TokageItLab** [comment]
For now, note that there is comment that reduz's lgtm on contributors chat.
https://chat.godotengine.org/channel/animation?msg=eFmBbXF992AHvW5DM

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**